### PR TITLE
⚠️ Use metadata-only client when retrieving CRDs

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by vincepri** [comment]
Marking as ⚠️ for
```
  Incompatible changes:
  - ConvertReferenceAPIContract: changed from func(context.Context, sigs.k8s.io/controller-runtime/pkg/client.Client, *k8s.io/api/core/v1.ObjectReference) error to func(context.Context, github.com/go-logr/logr.Logger, sigs.k8s.io/controller-runtime/pkg/client.Client, *k8s.io/client-go/rest.Config, *k8s.io/api/core/v1.ObjectReference) error
```

File: util/conversion/conversion.go [hunk 1]
```diff
@@ -54,36 +56,34 @@ var (
...
-		return errors.Errorf("cannot find any versions matching contract %q for CRD %v", contract, crd.Name)
+		return errors.Errorf("cannot find any versions matching contract %q for CRD %v", contract, metadata.Name)
 	}
 
 	// Pick the latest version in the slice and validate it.
 	kubeVersions := util.KubeAwareAPIVersions(strings.Split(supportedVersions, "_"))
 	sort.Sort(kubeVersions)
 	chosen := kubeVersions[len(kubeVersions)-1]
 
-	// Validate that the picked version is actually in the CRD spec.
```
Comment (fabriziopandini) [inline util/conversion/conversion.go:75]: Working with metadata we are losing the validation that the picked version is an actual version, but I think this is an acceptable trade-off

**Review by fabriziopandini — COMMENTED** [review]
approve pending @yastij review

**Review by yastij — APPROVED** [review]
/lgtm
/approve

**Pull request merged**